Copyright (c) 2013 MontaVista Software, Inc.  All rights reserved.

Save the baud rate if no baud has been provided.

Index: util-linux-2.22.2/term-utils/agetty.c
===================================================================
--- util-linux-2.22.2.orig/term-utils/agetty.c
+++ util-linux-2.22.2/term-utils/agetty.c
@@ -702,6 +702,9 @@ static void parse_args(int argc, char **
 				parse_speeds(op, v);
 			else
 				op->speeds[op->numspeed++] = bcode("9600");
+		} else {
+			/* If no speed was specified, don't change it. */
+			op->flags |= F_KEEPSPEED;
 		}
 	}
 
